backend/app/api/endpoints/media.py

```diff
@@ -0,0 +1,215 @@
+                "id": "b0feaabc-da2b-442e-9b3e-55c11c2c2ff2",
+                "name": "cat.jpg",
+            },
+        }
+    }
+
+
+class PaginationMetadata(BaseModel):
+    """Pagination metadata for list responses"""
+
+    total_count: int
+    page: int
+    page_size: int
+    total_pages: int
+
+
+class MediaListResponse(BaseModel):
+    """Response model for paginated media list"""
+
+    data: list[Media]
+    pagination: PaginationMetadata
+
+
+@router.get(
+    "",
+    response_model=MediaListResponse,
+    responses={
+        status.HTTP_200_OK: {
+            "description": "List of available media",
+            "content": {
+                "application/json": {
+                    "examples": MEDIA_RESPONSE_LIST_EXAMPLES,
+                }
+            },
+        },
+        status.HTTP_400_BAD_REQUEST: {"description": "Invalid query parameters"},
+    },
+)
+async def list_media(
+    page: int = Query(default=1, ge=1, description="Page number (starts from 1)"),
+    page_size: int = Query(default=20, ge=1, le=100, description="Number of items per page (max 100)"),
+    start_date: date | None = Query(
+        default=None, description="Filter media created on or after this date (YYYY-MM-DD)"
+    ),
+    end_date: date | None = Query(default=None, description="Filter media created on or before this date (YYYY-MM-DD)"),
+) -> list[Media]:
+    """List available media items with pagination and optional date filtering"""
+    _ = page, page_size
+    if start_date and end_date and start_date > end_date:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="Start date cannot be after end date",
+        )
+    raise NotImplementedError
+
+
+@router.get(
+    "/{media_id}",
+    response_model=Media,
+    responses={
+        status.HTTP_200_OK: {"description": "Media found"},
+        status.HTTP_400_BAD_REQUEST: {"description": "Invalid media ID"},
+        status.HTTP_404_NOT_FOUND: {"description": "Media not found"},
+    },
+)
+async def get_media(
+    media_id: Annotated[UUID, Depends(get_media_id)],
+) -> Media:
+    """Get detailed information about a specific media item"""
+    _ = media_id
+    raise NotImplementedError
+
+
+@router.get(
+    "/{media_id}/full",
+    responses={
+        status.HTTP_200_OK: {
+            "description": "Media found",
+            "content": {  # TODO: Indicate all supported binary formats
+                "image/jpeg": {"example": "...binary data..."},
+                "image/png": {"example": "...binary data..."},
+            },
+        },
+        status.HTTP_400_BAD_REQUEST: {"description": "Invalid media ID"},
+        status.HTTP_404_NOT_FOUND: {"description": "Media not found"},
+    },
+)
+async def get_full(
+    media_id: Annotated[UUID, Depends(get_media_id)],
+) -> FileResponse:
+    # In Geti Classic media microservice, image was streamed from object storage, so `StreamingResponse` is appropriate.
```

Comment:
Let's try to keep the documentation self-contained avoiding references to other Geti projects, unless necessary